File: 0094. Binary Tree Inorder Traversal/solution_2.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */

/**
 * @param {TreeNode} root
 * @return {number[]}
 */

var inorderTraversal = function (root) 
{
    const values = []
    const stack = []

    let node = root

    while (node != null || stack.length != 0)
    {
        while (node)
        {
            stack.push(node)
            node = node.left
        }

        node = stack.pop();

        values.push(node.val)

        node = node.right
    }

    return values
};

File: 0094. Binary Tree Inorder Traversal/solution_2.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
 
 public class Solution
{
    public IList<int> InorderTraversal(TreeNode root)
    {
        List<int> values = new List<int>();
        List<TreeNode> stack = new List<TreeNode>();

        TreeNode node = root;

        while (node != null || stack.Count != 0)
        {
            while (node != null)
            {
                stack.Add(node);
                node = node.left;
            }

            node = stack.Last();
            stack.RemoveAt(stack.Count - 1);

            values.Add(node.val);

            node = node.right;
        }

        return values;
    }
}